node/examples/repl_server.js

```diff
@@ -89,19 +91,24 @@ function ReplSession(repler, options) {
     self.logger = options.logger;
 }
 
+util.inherits(ReplSession, EventEmitter);
+
 ReplSession.prototype.start = function start(stream) {
-    // function startReplSession() {
-    //     setImmediate(startReplSession);
-    // }
     var self = this;
-    self.repler.open(stream);
+    self.once('resize', function onResize(size) {
+        self.repler.open(stream, {
+            width: size.cols,
+            height: size.rows
+        });
+    });
 };
 
 ReplSession.prototype.control = function control(stream) {
     var self = this;
-    stream.on('data', function(obj) {
+    stream.on('data', function onControlData(obj) {
         if (obj.op === 'resize') {
             self.logger.info('resize to', obj);
+            self.emit('resize', obj);
```

Comment:
works for me, land at will

On Mon, Apr 6, 2015 at 3:57 PM, Rob Skillington <notifications@github.com>
wrote:

> In node/examples/repl_server.js
> <https://github.com/uber/tchannel/pull/334#discussion_r27838635>:
>
> >          if (obj.op === 'resize') {
> >              self.logger.info('resize to', obj);
> > +            self.emit('resize', obj);
>
> Hmm, it makes the whole thing a little more stateful and cluttered, having
> to use a boolean and reset the boolean to only open it on the first resize.
> Considering this is REPL and is invoked only when in use perhaps this setup
> is okay?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/uber/tchannel/pull/334/files#r27838635>.
>